features/nfc/nfc/NFCEEPROMDriver.h

```diff
@@ -0,0 +1,200 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_NFC_EEPROM_DRIVER_H
+#define MBED_NFC_EEPROM_DRIVER_H
+
+#include <stdint.h>
+
+#include "NFCDefinitions.h"
+#include "NFCTarget.h"
+#include "NFCEEPROMDriver.h"
+
+namespace mbed {
+namespace nfc {
+
+    /**
+     * @addtogroup nfc
+     * @{
+     */
+    
+    /**
+     * The abstraction for a NFC EEPROM driver.
+     * Implementers need to derive from this class and implement its methods.
+     */ 
+    class NFCEEPROMDriver  {
+    public:
+        /**
+         * Construct a NFCEEPROM driver instance.
+         */
+        NFCEEPROMDriver();
+        
+        /**
+         * The NFCEEPROMDriver delegate.
+         * Methods in this class are called by the driver on completion of long-running operations.
+         */
+        struct Delegate {
+            /**
+             * Completion of session start operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void on_session_started(bool success) = 0;
+
+            /**
+             * Completion of session end operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void on_session_ended(bool success) = 0;
+
+            /**
+             * Completion of read operation.
+             * 
+             * @param[in] count number of bytes actually read
+             */  
+            virtual void on_bytes_read(size_t count) = 0;
+
+            /**
+             * Completion of write operation.
+             * 
+             * @param[in] count number of bytes actually written
+             */  
+            virtual void on_bytes_written(size_t count) = 0;
+
+            /**
+             * Completion of size retrieval operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             * @param[out] the current addressable memory size
+             */  
+            virtual void on_size_read(bool success, size_t size) = 0;
+
+            /**
+             * Completion of size setting operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void on_size_written(bool success) = 0;
+
+            /**
+             * Completion of erasing operation.
+             * 
+             * @param[in] count number of bytes actually erased
+             */  
+            virtual void on_bytes_erased(size_t count) = 0;
+
+            /**
+             * Signal the user that the process_events() need to be called
+             * 
+             * @note this function can be called in interrupt context
+             */
+            virtual void on_event() = 0;
```

Comment:
We can make this simpler by handing over the event queue which you already have in the nfc target

See:
https://github.com/ARMmbed/mbed-nfc-m24sr/pull/1/commits/ec0d19146ff745940eee8d32bb839b10cbc88984#diff-3856d36332fc549ca6b08cafbfcac790